src/shared/components/query/DataTypePrioritySelector.spec.tsx

```diff
@@ -0,0 +1,108 @@
+            }] as MolecularProfile[];
+            assert.deepEqual(profileAvailability(profiles), {mutation:false, cna:false});
+        });
+        it("returns correct profile availability result in case of two profiles", ()=>{
+            let profiles = [{
+                molecularAlterationType: AlterationTypeConstants.MUTATION_EXTENDED,
+                showProfileInAnalysisTab: true
+            },{
+                molecularAlterationType: AlterationTypeConstants.COPY_NUMBER_ALTERATION,
+                showProfileInAnalysisTab: true
+            }] as MolecularProfile[];
+            assert.deepEqual(profileAvailability(profiles), {mutation:true, cna:true});
+
+            profiles = [{
+                molecularAlterationType: AlterationTypeConstants.MUTATION_EXTENDED,
+                showProfileInAnalysisTab: false
+            },{
+                molecularAlterationType: AlterationTypeConstants.COPY_NUMBER_ALTERATION,
+                showProfileInAnalysisTab: true
+            }] as MolecularProfile[];
+            assert.deepEqual(profileAvailability(profiles), {mutation:false, cna:true});
+
+            profiles = [{
+                molecularAlterationType: AlterationTypeConstants.MUTATION_EXTENDED,
+                showProfileInAnalysisTab: true
+            },{
+                molecularAlterationType: AlterationTypeConstants.COPY_NUMBER_ALTERATION,
+                showProfileInAnalysisTab: false
+            }] as MolecularProfile[];
+            assert.deepEqual(profileAvailability(profiles), {mutation:true, cna:false});
+
+            profiles = [{
+                molecularAlterationType: AlterationTypeConstants.MUTATION_EXTENDED,
+                showProfileInAnalysisTab: false
+            },{
+                molecularAlterationType: AlterationTypeConstants.COPY_NUMBER_ALTERATION,
+                showProfileInAnalysisTab: false
+            }] as MolecularProfile[];
+            assert.deepEqual(profileAvailability(profiles), {mutation:false, cna:false});
+        });
+        it("returns correct profile availability result in case of several profiles", ()=>{
+            let profiles = [{
+                molecularAlterationType: AlterationTypeConstants.MUTATION_EXTENDED,
+                showProfileInAnalysisTab: true
+            },{
+                molecularAlterationType: AlterationTypeConstants.MUTATION_EXTENDED,
+                showProfileInAnalysisTab: false
+            },{
+                molecularAlterationType: AlterationTypeConstants.COPY_NUMBER_ALTERATION,
+                showProfileInAnalysisTab: false
+            }] as MolecularProfile[];
+            assert.deepEqual(profileAvailability(profiles), {mutation:true, cna:false});
+        });
+    });
+    describe("radioButtons",()=>{
+        it("shows the right buttons when theres mutation and cna profiles available",()=>{
+            const buttonLabels = stringListToSet(radioButtons({mutation:true, cna:true}).map(x=>x.props.label));
```

Comment:
i would use enzyme here.  it's more semantic and doing so in general would allow easier extension of the tests if necessary.

Reply:
one could extract the label logic into a pure function so as to avoid the difficult of testing component itself.  but actually i think it's better to test component because it makes it easier to test other behaviors of the component, should there be any. let me know if that makes sense.

Reply:
thats more complicated than necessary imo

Reply:
when writing this test I'm thinking, what is the behavior thats really at risk of regression? I dont think theres any reason to test for a regression in what specifically are the elements, what relative position, etc.